storage/test/tuples.go

```diff
@@ -457,3 +458,199 @@ func TuplePaginationOptionsTest(t *testing.T, datastore storage.OpenFGADatastore
 		}
 	})
 }
+
+func ReadStartingWithUserTest(t *testing.T, datastore storage.OpenFGADatastore) {
+
+	require := require.New(t)
+	ctx := context.Background()
+
+	tuples := []*openfgapb.TupleKey{
+		tuple.NewTupleKey("document:doc1", "viewer", "user:jon"),
+		tuple.NewTupleKey("document:doc2", "viewer", "group:eng#member"),
+		tuple.NewTupleKey("document:doc3", "editor", "user:jon"),
+		tuple.NewTupleKey("folder:folder1", "viewer", "user:jon"),
+	}
+
+	t.Run("returns results with two user filters", func(t *testing.T) {
+		storeID, err := id.NewString()
+		require.NoError(err)
+
+		err = datastore.Write(
+			ctx,
+			storeID,
+			nil,
+			tuples,
+		)
+		require.NoError(err)
+
+		iter, err := datastore.ReadStartingWithUser(
+			ctx,
+			storeID,
+			storage.ReadStartingWithUserFilter{
+				ObjectType: "document",
+				Relation:   "viewer",
+				UserFilter: []*openfgapb.ObjectRelation{
+					{
+						Object: "user:jon",
+					},
+					{
+						Object:   "group:eng",
+						Relation: "member",
+					},
+				},
+			},
+		)
+		require.NoError(err)
+
+		var objects []string
+		for {
+			tp, err := iter.Next()
```

Comment:
This iterator code is used here and below. We should create a helper so we avoid code duplication.